Makefile

```diff
@@ -0,0 +1,136 @@
+# Basic Makefile for Golang project
+# Includes GRPC Gateway, Protocol Buffers
+SERVICE		?= $(shell basename `go list`)
+VERSION		?= $(shell git describe --tags --always --dirty --match=v* 2> /dev/null || cat $(PWD)/.version 2> /dev/null || echo v0)
+PACKAGE		?= $(shell go list)
+PACKAGES	?= $(shell go list ./...)
+FILES		?= $(shell find . -type f -name '*.go' -not -path "./vendor/*")
+BUILD_DIR	?= build
+
+# Binaries
+PROTOC		?= protoc
+
+.PHONY: help clean fmt lint vet test test-cover build build-docker all
+
+default: help
+
+# show this help
+help:
+	@echo 'usage: make [target] ...'
+	@echo ''
+	@echo 'targets:'
+	@egrep '^(.+)\:\ .*#\ (.+)' ${MAKEFILE_LIST} | sed 's/:.*#/#/' | column -t -c 2 -s '#'
+
+# clean, format, build and unit test
+all:
+	make clean-all
+	make gofmt
+	make build
+	make test
+
+# build and install go application executable
+install:
+	go install -v ./...
+
+# Print useful environment variables to stdout
+env:
+	echo $(CURDIR)
+	echo $(SERVICE)
+	echo $(PACKAGE)
+	echo $(VERSION)
+
+# go clean
+clean:
+	go clean
+
+# remove all generated artifacts and clean all build artifacts
+clean-all:
+	go clean -i ./...
+	rm -fr rpc
+
+# fetch and install all required tools
+tools:
+	go get -u golang.org/x/tools/cmd/goimports
+	go get -u github.com/golang/lint/golint
+	go get -u github.com/matryer/moq
+	# Reqire when adding proto and grpc compilation
+	# go get -u github.com/golang/protobuf/protoc-gen-go
+	# go get -u github.com/mwitkow/go-proto-validators/protoc-gen-govalidators
+	# go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
+	# go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger
+
+# format the go source files
+fmt:
+	go fmt ./...
```

Comment:
`go fmt` is not necessary because `goimports` is a superset of `go fmt` and does everything `go fmt` do.